### PR TITLE
Design Prototype: Deprecate IRuntime Request

### PR DESCRIPTION
**Pull request by anthony-murphy**
This change is a prototype where we deprecate the request method on IRuntime, and move to an taking a default entrypoint instead.

### CODE REVIEW

File: packages/loader/container-loader/src/containerContext.ts [hunk 1]
```diff
@@ -240,10 +240,18 @@ export class ContainerContext implements IContainerContext {
...
+    /**
+     * @deprecated - use getRuntimeEntryPoint instead
+     */
     public async request(path: IRequest): Promise<IResponse> {
         return this.runtime.request(path);
     }
 
+    public async getRuntimeEntryPoint(): Promise<IFluidObject> {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return (this.runtime as any).getEntryPoint?.();
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/containerContext.ts:252]: i didn't bother updating definitions to just using a cast in the prototype to access the entry point

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -539,6 +538,43 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         containerScope: IFluidObject = context.scope,
         existing?: boolean,
     ): Promise<ContainerRuntime> {
+        let router: IFluidRouter | undefined;
+        const requestToEntryPoint = async (cr: IContainerRuntime)=>{
```
Comment (anthony-murphy) [inline packages/runtime/container-runtime/src/containerRuntime.ts:542]: for old load, build an entrypoint that does the old request pattern

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -1113,32 +1153,30 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
-                return create404Response(request);
-            }
-            if (this.requestHandler !== undefined) {
-                return this.requestHandler(parser, this);
+    async request(request: IRequest): Promise<IResponse> {
+        const parser = RequestParser.create(request);
+        const id = parser.pathParts[0];
+
+        // todo - this should move somewhere else so we
+        // don't need to request to get it
```
Comment (anthony-murphy) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1163]: current idea is to make it the default entry point of runtimes that are summarizers

File: packages/runtime/runtime-definitions/src/dataStoreContext.ts [hunk 4]
```diff
@@ -82,11 +82,6 @@ export interface IContainerRuntimeBase extends
      */
     setFlushMode(mode: FlushMode): void;
 
-    /**
-     * Executes a request against the container runtime
-     */
-    request(request: IRequest): Promise<IResponse>;
```
Comment (anthony-murphy) [inline packages/runtime/runtime-definitions/src/dataStoreContext.ts:88]: ContainerRuntimeBase no longer has request, but it does have getEntryPoint which still isn't ideal. ideally the datastore is give handles to any dependencies at construction time which the data store persists, and it never need to reach back up

File: packages/test/test-end-to-end-tests/src/test/mapEndToEndTests.spec.ts [hunk 5]
```diff
@@ -25,6 +25,16 @@ const testContainerConfig: ITestContainerConfig = {
     registry,
 };
 
+const getDefaultTestObject = async (container: Container): Promise<ITestFluidObject>=>{
```
Comment (anthony-murphy) [inline packages/test/test-end-to-end-tests/src/test/mapEndToEndTests.spec.ts:28]: example of back compat host code in a test. it works in most cases, but didn't make it complete

File: packages/runtime/runtime-definitions/src/dataStoreContext.ts [hunk 6]
```diff
@@ -71,6 +69,8 @@ export interface IContainerRuntimeBase extends
     readonly logger: ITelemetryBaseLogger;
     readonly clientDetails: IClientDetails;
 
+    getEntryPoint(): Promise<IFluidObject>;
```
Comment (vladsud) [inline packages/runtime/runtime-definitions/src/dataStoreContext.ts:72]: I'd think that the only place it should show up is IRuntime, no? It's purpose is communicating up (to loader), not down (to data stores).

Comment (anthony-murphy) [reply packages/runtime/runtime-definitions/src/dataStoreContext.ts:72]: you're probably right that this didn't need to push down to here. We'll need the same thing for the datastore runtime as well

**Pull request closed without merging**